sw/device/silicon_creator/rom_ext/rom_ext_common.ld

```diff
@@ -23,6 +23,9 @@ __DYNAMIC = 0;
  */
 ENTRY(_rom_ext_start_boot)
 
+/* DV Log offset (has to be different to other boot stages). */
+_dv_log_offset = 0x10000;
```

Comment:
I assume this is a nop? We want to make sure there is no DV log functionality in the rom/rom_ext production version. It would be good to clarify in the note that the offset is not used.

Reply:
I don't think we have a way of doing that at all yet. To conditionally "compile" things out of the linker script we'd need to use the C preprocessor on the linker files, which we can't wire up until Meson is gone.

This code is also present in the mask ROM.

@timothytrippel how would you like to deal with it.

Reply:
Just assigning the symbol here is essentially a nop, since the mask ROM configures the PMP to lock out this region, so even if any ROM_EXT code attempts to LOG something here it won't be able to, unless the ROM_EXT itself re-enables writes to this region.

The short-term plan was to have the OTTF (which currently runs at the ROM_EXT stage) to re-able this region for DV logging. However, this only works if the OTTF (and subsequently the test) run at the ROM_EXT stage, where it will have the required privileges to do so. Thinking forward as we aim to support running OTTF-launched tests at different stages (ROM_EXT vs BL0) for key-management purposes it would be nice to have a way to enable this for non-production versions, but disable for production version. By then, meson will be gone so the issue you describe @mcy shouldn't be a problem anymore. However, this is obviously dangerous so we will probably need to discuss how to handle enabling writes to the region at some point.

For now, you could drop a comment here as @moidx suggests, and say how if a test runs at BL0, booted by a ROM_EXT linked with this script, DV sim LOGs won't work until #10863 and #10712 are resolved.